src/reducers/reducer.js

```diff
@@ -1086,21 +1104,46 @@ export const onSetWCConnectionFailed = (state, { payload }) => ({
  * @param {Object} action.payload The network settings emitted in saga
  * @see updateNetworkSettings
  */
-export const onNetworkSettingsUpdate = (state) => ({
+export const onNetworkSettingsUpdateRequest = (state) => ({
   ...state,
   networkSettingsStatus: NETWORKSETTINGS_STATUS.LOADING,
 });
 
 /**
  * @param {Object} action.payload The network settings emitted in saga
- * @see updateNetworkSettings
+ * @see networkSettingsUpdateState customNetwork
+ */
+export const onNetworkSettingsUpdateState = (state, { payload }) => ({
+  ...state,
+  networkSettings: payload,
+});
+
+/**
+ * @param {Object} action.payload The network settings emitted in saga
+ * @see networkSettingsPersistStore customNetwork
  */
-export const onNetworkSettingsUpdateSucess = (state, { payload }) => ({
+export const onNetworkSettingsPersistStore = (state, { payload }) => ({
   ...state,
   networkSettings: payload,
```

Comment:
Why do you update the `networkSettings` again? Didn't you update it in the `onNetworkSettingsUpdateState` method?

Reply:
Oh! I do it here because the `updateNetworkSettings` effect calls this persistence, and this is the opportunity to update the redux state. The `onNetworkSettingsUpdateRequest` is triggered only by the `initNetworkSettings` effect when the wallet initializes and a network settings is found persisted in the app storage.